bitbots_localization/include/bitbots_localization/Resampling.h

```diff
@@ -0,0 +1,120 @@
+//
+// Created by judith on 17.04.19.
+//
+
+#ifndef IMPORTANCERESAMPLINGWE_H
+#define IMPORTANCERESAMPLINGWE_H
+
+#include <assert.h>
+#include "particle_filter/ImportanceResampling.h"
+#include "particle_filter/CRandomNumberGenerator.h"
+
+// ImportanceResampling with explorers
+
+template <class StateType>
+class ImportanceResamplingWE
+        : public particle_filter::ImportanceResampling<StateType> {
+    /**
+     * A ParticleList is an array of pointers to Particles.
+     */
+    typedef std::vector<particle_filter::Particle<StateType>*> ParticleList;
+
+public:
+    /**
+     * The constructor of this base class inits some members.
+     */
+    ImportanceResamplingWE<StateType>(int explorer_count,
+                                      std::shared_ptr<particle_filter::StateDistribution<StateType>>
+                                      distribution);
+
+    /**
+     * The destructor is empty.
+     */
+    virtual ~ImportanceResamplingWE();
+
+    /**
+     * This is the main method of ImportanceResampling. It takes two references
+     * to particle lists. The first reference refers to the old particle list,
+     * the second to the new one.
+     * @param source the source list to draw new particles from.
+     * @param destination the destination list where to put the copies.
+     */
+    void
+    resample(const ParticleList& source, const ParticleList& destination) const;
+
+    /**
+     * Sets the Random Number Generator to use in resample() to generate
+     * uniformly distributed random numbers.
+     */
+    void setExplorerCount(int explorer_count);
+
+    int getExplorerCount();
+
+private:
+    int explorer_count_;
+    std::shared_ptr<particle_filter::StateDistribution<StateType>>
+            distribution_;
+
+    particle_filter::CRandomNumberGenerator m_RNG;
+};
+
+
+template <class StateType>
+ImportanceResamplingWE<StateType>::ImportanceResamplingWE(int explorer_count,
+                                                          std::shared_ptr<particle_filter::StateDistribution<StateType>>
+                                                          distribution) :
+        explorer_count_(explorer_count),
+        distribution_(distribution) {
+    particle_filter::ImportanceResampling<StateType>();
+}
+
+template <class StateType>
+ImportanceResamplingWE<StateType>::~ImportanceResamplingWE() {}
```

Comment:
you should use `= default` to define the trivial destructor or is there any reason why not?